File: codeforces/way_too_long_words.py
n = int(input())
for _ in range(n):
    word = input()
    if len(word) <= 10:
        print(word)
    else:
        #print(word[0] + str(len(word[1:-1])) + word[-1])
        print(word[0] + str(len(word) - 2) + word[-1])

File: codeforces/java/src/WayTooLongWords.java
// https://codeforces.com/problemset/problem/71/A
import java.util.Scanner;

public class WayTooLongWords {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();

        for (int i = 0; i < n; i++) {
            String word = sc.nextLine();
            System.out.println(word.length() <= 10 ? word : word.charAt(0) + Integer.toString(word.length() - 2) + word.charAt(word.length() - 1));
        }
    }
}
